html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/SponsorDefined.pm

```diff
@@ -0,0 +1,130 @@
+package captiveportal::PacketFence::DynamicRouting::Module::Authentication::SponsorDefined;
+
+=head1 NAME
+
+captiveportal::DynamicRouting::Module::Authentication::SponsorDefined
+
+=head1 DESCRIPTION
+
+Sponsoring module
+
+=cut
+
+use Moose;
+use pf::Authentication::constants;
+use pf::config qw(%Config);
+use pf::log;
+use Date::Format qw(time2str);
+use pf::activation;
+use pf::web::guest;
+use pf::util qw(normalize_time);
+
+extends 'captiveportal::DynamicRouting::Module::Authentication::Sponsor';
+
+has '+source' => (isa => 'pf::Authentication::Source::SponsorEmailSource');
+has 'sponsor' => (is => 'rw');
+
+
+
+=head2 _build_required_fields
+
+Build the required fields based on the PID field, the custom fields and the mandatory fields of the source
+
+=cut
+
+sub _build_required_fields {
+    my ($self) = @_;
+    my @fields = ((grep {$_ ne 'sponsor'} @{$self->SUPER::_build_required_fields()}));
+
+    return \@fields;
+}
+
+sub do_sponsor_registration {
```

Comment:
I'd dry this up a bit so it isn't a copy/paste of the one in sponsor

I can look into that post 7.0